library/lua/enable-util.lua

```diff
@@ -0,0 +1,54 @@
+
+local _ENV = mkmodule("enable-util")
+
+local repeatUtil = require('repeat-util')
+
+function assignPersistentGlobal(global_key, global_store, key, value)
+    if global_store[key] == value then
+        return
+    end
+    global_store[key] = value
+    dfhack.persistent.saveSiteData(global_key, global_store)
+end
+
+function enableEvery(flags, global_store, global_key, time, unit, action)
+
+    local function assign_persistent(key,value)
+        assignPersistentGlobal(global_key, global_store, key, value)
+    end
+
+    dfhack.onStateChange[global_key] = function(sc)
+        if sc == SC_MAP_UNLOADED then
+            global_store.enabled = false
+            -- repeat-util will cancel on its own
+            return
+        end
+
+        if sc ~= SC_MAP_LOADED or df.global.gamemode ~= df.game_mode.DWARF then
```

Comment:
especially as adventure mode is expected soon, this will need to become more flexible.